File: Napakalaki Ruby/lib/treasure.rb
# encoding: UTF-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'card'

module Napakalaki

  class Treasure
    
    include Card
    
    attr_reader :name,:goldCoins,:minBonus,:maxBonus,:Type;

    def initialize (name, goldCoins, min, max, treasurekind)

      @name = name
      @goldCoins = goldCoins
      @minBonus = min
      @maxBonus = max
      @Type = treasurekind

    end

    def to_s

      "#{@name} ,Type #{@Type} ,Max.Bonus #{@maxBonus} ,Min.Bonus #{@minBonus} ,Gold coins #{@goldCoins}"

    end

    def getType
    
      return @Type
      
    end
    
    def getBasicValue
    
      return @minBonus

    end

    def getSpecialValue

      return @maxBonus

    end
    
  end

end

File: Napakalaki Java/src/NapakalakiGame/Treasure.java
/*
 */
package NapakalakiGame;

/**
 *
 */
public class Treasure implements Card{
    
    private String name = new String();
    private int goldCoins;
    private int minBonus;
    private int maxBonus;
    private TreasureKind Type;
    
    public Treasure(String nam, int gold, int min, int max, TreasureKind tk){
        
        name = nam;
        goldCoins = gold;
        minBonus = min;
        maxBonus = max;
        Type = tk;
    }
    
    
    public String getName(){
        
        return name;
    }
    
    public int getGoldCoins(){
    
        return goldCoins;
    }
    
    public int getMinBonus(){
    
        return minBonus;
    }
    
    public int getMaxBonus(){
    
        return maxBonus;
    }
    
    public TreasureKind getType(){
    
        return Type;
    }
    
    public String toString(){
    
        return name + ", Type = "  + Type + ", Max.Bonus = " + maxBonus + ", Min.Bonus = " + minBonus + ", Gold coins = "  + goldCoins;
    }
    
    public String TypetoString(){
    
        return "" + Type;
    }

    @Override
    public int getBasicValue() {
        
        return getMinBonus();
    }

    @Override
    public int getSpecialValue() {

        return getMaxBonus();
    }
}